Assignment 5.Rmd

```diff
@@ -1,37 +1,72 @@
 
 #Install & call libraries
-```{r}
-install.packages("party", "rpart")
+```{r eval = FALSE}
+#Install needed packages
+#I've done this, so eval = FALSE so it doesn't run each time
+install.packages("party")
+install.packages("rpart")
+```
 
+
+```{r message = FALSE}
+#Import rpart
 library(rpart)
+
+#Import party!
 library(party)
 ```
 
 ## Part I
 ```{r}
-D1 <- 
+D1 <- read.csv("intelligent_tutor.csv")
 ```
 
-##Classification Tree
-First we will build a classification tree to predict which students ask a teacher for help, which start a new session, or which give up, based on whether or not the student completed a session (D1$complete) and whether or not they asked for hints (D1$hint.y). 
-```{r}
 
-c.tree <- rpart(action ~ hint.y + complete, method="class", data=D1) #Notice the standard R notion for a formula X ~ Y
+Codebook
+* id - student id
+* prior_prob_count - The number of problems a student has done in the system prior to the surrent session  
+* score - The score the student achieved in the current session  
+* hints - The number of hints the student requested in the current session  
+* hint.y - Whether or not the student asked for hints in the current session  
+* complete - Whether or not the student completed the current session  
+* action - The action suggested by the system to a teacher about a given student based on their performance
+
+##Classification Tree - CART
+First we will build a classification tree to predict which students ask a teacher for help, which start a new session, or which give up, based on whether or not the student completed a session `(D1$complete)` and whether or not they asked for hints `(D1$hint.y)`. 
+```{r}
+#Build the classification tree
+c.tree <- rpart(action ~ hint.y + complete, method = "class", data = D1) #Notice the standard R notion for a formula X ~ Y
 
 #Look at the error of this tree
 printcp(c.tree)
 
 #Plot the tree
-post(c.tree, file = "tree.ps", title = "Session Completion Action: 1 - Ask teacher, 2 - Start new session, 3 - Give up")
+#This saves it to a file called tree.ps
+#post(c.tree, filename = "tree.ps", title = "Session Completion Action: 1 - Ask teacher, 2 - Start new session, 3 - Give up")
+
+#This prints our tree to the Rmd file, which is what we want
+post(c.tree, filename = "", title = "Session Completion Action: 1 - Ask teacher, 2 - Start new session, 3 - Give up")
 
 ```
+Printcp output gives error rates at each split:
+
+* Root node error - the number of misclassifications/total n in the top-level (root) node, before any splits have occurred
+* CP - looks like the marginal decrease in `rel error` with each split
+* nsplit - number of splits in the tree
+* rel error - When multiplied by the root node error, gives the resubstitution error rate (error rate computed against the training sample, so probably crap)
+* xerror - When multiplied by the root node error, gives the cross-validated error rate using 10-fold CV.
+* xstd - ??? - probably K-fold error rate standard error?
```

Comment:
xstd is the standard error of xerror, which is an error estimated from a ten-fold cross-validation.